pilot/pkg/xds/discovery.go

```diff
@@ -217,7 +216,7 @@ func (s *DiscoveryServer) getNonK8sRegistries() []serviceregistry.Instance {
 			},
 		}
 	}
-
+	nonK8sRegistries := make([]serviceregistry.Instance, 0, len(registries))
```

Comment:
This is not accurate

Reply:
This is ok to set noK8sregistries capacity to total too worst case. hence, kept it because registries = agg.GetRegistries() returns total [registries](https://github.com/istio/istio/blob/master/pilot/pkg/serviceregistry/aggregate/controller.go#L44) and hence its ok to use the length .